hi-ml/testhiml/testhiml/test_training_runner.py

```diff
@@ -324,6 +325,12 @@ def test_init_inference(
                 assert not training_runner_hello_world_with_checkpoint.container.model._on_extra_val_epoch
 
                 training_runner_hello_world_with_checkpoint.init_inference()
+                if run_extra_val_epoch:
+                    assert (
+                        caplog.messages[-3]
+                        == "Preparing to run an extra validation epoch to evaluate the model on the validation set."
+                    )
+                assert caplog.messages[-2] == "Preparing runner for inference."
```

Comment:
If the call order was swapped back to what it was before your change, would this test fail? Maybe better check if the inference bag sizes are changed as expected?

Reply:
yes, because of the order in the messages, I know it's not ideal because the logs order doesn't guarantee the right functions call order 
note that the change in bag size is specific to MIL container, also get_data_module is mocked here, do you know if there's a way to check if a mock has been called before another?